### ISSUE TITLE
OSV files for Sentinel-1 only from 2021?

### ISSUE THREAD

**Issue by florianbeyer** [johntruckenbrodt/pyroSAR#139]
I am aware of the current discussion about the orbit files. [SNAP Forum](https://forum.step.esa.int/t/orbit-file-timeout-march-2021/28621)
If I correctly identify the platform used to receive the Orbite files, you are already using the new address, correct?

However, is there a solution to get all orbite files starting at the beginning of the S1 mission?
Even the files from 2020 seem not to be available with your functions...

Is there a possibility to search for orbit files on both adresses https://scihub.copernicus.eu and https://qc.sentinel1.eo.esa.int/?

- which operating system are you using?  
Windows 10

- which environment is pyroSAR running in?  
Anaconda environment (python3.8)

- which version of pyroSAR are you using?  
0.12.1

-  which function of pyroSAR did you call with which parameters?  
```
with OSV(osvdir) as osv:
    osv_files = osv.catch(sensor='S1A', osvtype='POE',
                     start='20150101T000000', stop='20211231T000000')
    osv.retrieve(osv_files)
```

- if applicable, which version of SNAP or GAMMA are you using in pyroSAR?

- the full error message

> searching for new POE files
> https://scihub.copernicus.eu/gnss/search/?q=producttype:AUX_POEORB platformname:Sentinel-1 filename:S1A* beginPosition:[2015-01-01T00:00:00Z TO 2021-12-31T00:00:00Z] endPosition:[2015-01-01T00:00:00Z TO 2021-12-31T00:00:00Z]&format=json
>found 49 OSV results
>downloading 1 file
> deleting 0 RES files

**Comment by johntruckenbrodt** [comment]
Hi @florianbeyer. Really annoying that this is still a thing. The new scihub GNSS page should contain all files by now but apparently this is not the case. This is the one pyroSAR is querying. The page  https://qc.sentinel1.eo.esa.int/ is offline for good. The only alternative I know is https://aux.sentinel1.eo.esa.int. This page cannot be used by pyroSAR yet. I was hoping this would not be necessary with the new GNSS site. I'll try to find time this week to integrate it.

**Comment by florianbeyer** [comment]
Thanks for the update @johntruckenbrodt. I really lost track about the current status. I guess we should wait until everything is uploaded on the new page...

**Comment by johntruckenbrodt** [comment]
Yep, really strange situation and not easy to keep track. Thanks for pointing out that the issue persists. Here's another resource: http://step.esa.int/auxdata/orbits/Sentinel-1/

**Comment by johntruckenbrodt** [comment]
I have now implemented a first try at downloading files from https://aux.sentinel1.eo.esa.int/.
The URL can be selected in [OSV.catch](https://pyrosar.readthedocs.io/en/latest/pyroSAR.html#pyroSAR.S1.OSV.catch) with a new argument `url_option`. `OSV.retrieve` is basically unchanged.  
@system123, @tyler-c2s this could be interesting for you too.

**Comment by elsadg** [comment]
Hi @johntruckenbrodt, thanks for adding this new option. I have pyroSAR version 0.12.1 and python version 3.9.2 and trying to test the new `url_option` in `OSV.catch` but it gives me an error below. Would you have any suggestions, please?

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-15-7cc177da9b5a> in <module>
      4 
      5 with OSV(osvdir) as osv:
----> 6     files = osv.catch(sensor='S1A', osvtype='POE',
      7                       start='20170101T000000', stop='20180101T000000',
      8                       url_option = 2)

TypeError: catch() got an unexpected keyword argument 'url_option'
```

**Comment by johntruckenbrodt** [comment]
Hi @elsadg. The `url_option` argument is not yet part of a pyroSAR release. You need to clone the current master branch to use it. I had not published a new version with it because the situation is so unclear right now. It seems that the second site is also offline now so effectively there is again only one URL option in pyroSAR now.


Here the link to the docs of version 0.12.1:
https://pyrosar.readthedocs.io/en/v0.12.1/pyroSAR.html#pyroSAR.S1.OSV.catch

**Comment by elsadg** [comment]
Thanks for the clarification @johntruckenbrodt. So the only site that should be working is this: https://scihub.copernicus.eu/gnss/search/. However, for me that site does not work either (maybe this is just an issue I have due to proxy?). Any ideas?

Thanks a lot!

`ConnectTimeout: HTTPSConnectionPool(host='scihub.copernicus.eu', port=443): Max retries exceeded with url: /gnss/search/?q=producttype:AUX_POEORB%20platformname:Sentinel-1%20filename:S1A*%20beginPosition:%5B2017-01-01T00:00:00Z%20TO%202018-01-01T00:00:00Z%5D%20endPosition:%5B2017-01-01T00:00:00Z%20TO%202018-01-01T00:00:00Z%5D&format=json (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f267040ef40>, 'Connection to scihub.copernicus.eu timed out. (connect timeout=20)'))`

**Comment by johntruckenbrodt** [comment]
Yep, this is the one.

I just gave it a try myself. Going by your error message I went with this code:
```Python
from pyroSAR.S1 import OSV

start = '20170101T000000'
stop = '20180101T000000'

with OSV(timeout=20) as osv:
    files = osv.catch(sensor='S1A', start=start, stop=stop, url_option=1)
```
Works fine.
Could be just a temporary issue. Have you tried increasing the timeout?
If not then yes this could be a proxy issue.

**Comment by elsadg** [comment]
Thank you @johntruckenbrodt. I tried with the code you suggested and increased the timeout. Unfortunately, the issue persists so will need to investigate the proxy

**Comment by johntruckenbrodt** [comment]
Download from http://step.esa.int/auxdata/orbits/Sentinel-1/ is now implemented in c0eba03 via a third `url_option`.

**Comment by johntruckenbrodt** [comment]
It looks like the issue is solved. Feel free to reopen if there is more.
